src/main/java/org/folio/inventory/services/MoveItemIntoStatusService.java

```diff
@@ -54,6 +112,41 @@ public MoveItemIntoStatusService(ItemCollection itemCollection, Clients clients)
       .thenCompose(itemCollection::update);
   }
 
+  public CompletableFuture<Item> processMarkItemRestricted(WebContext context) {
+    final String itemId = context.getStringParameter("id", null);
+
+    return itemCollection.findById(itemId)
+      .thenCompose(ItemsValidator::refuseWhenItemNotFound)
+      .thenCompose(MarkAsRestrictedValidators::itemHasAllowedStatusToMarkAsRestricted)
+      .thenCompose(this::updateRequestStatusIfRequired)
+      .thenApply(item -> item.changeStatus(RESTRICTED))
+      .thenCompose(itemCollection::update);
+  }
+
+  public CompletableFuture<Item> processMarkItemUnavailable(WebContext context) {
+    final String itemId = context.getStringParameter("id", null);
+
+    return itemCollection.findById(itemId)
+      .thenCompose(ItemsValidator::refuseWhenItemNotFound)
+      .thenCompose(MarkAsUnavailableValidators::itemHasAllowedStatusToMarkAsUnavailable)
+      .thenCompose(this::updateRequestStatusIfRequired)
+      .thenApply(item -> item.changeStatus(UNAVAILABLE))
+      .thenCompose(itemCollection::update);
+  }
+
+
+  public CompletableFuture<Item> processMarkItemUnknown(WebContext context) {
+    final String itemId = context.getStringParameter("id", null);
+
+    return itemCollection.findById(itemId)
+      .thenCompose(ItemsValidator::refuseWhenItemNotFound)
+      .thenCompose(MarkAsUnknownValidators::itemHasAllowedStatusToMarkAsUnknown)
+      .thenCompose(this::updateRequestStatusIfRequired)
+      .thenApply(item -> item.changeStatus(UNKNOWN))
+      .thenCompose(itemCollection::update);
+  }
```

Comment:
We can extract a base method now, something like:

```java
  private CompletableFuture<Item> moveItemIntoStatus(WebContext context, ItemStatusName targetStatus,
    Function<Item, CompletableFuture<Item>> allowedStatusesValidator) {
    
    final String itemId = context.getStringParameter("id", null);
    return itemCollection.findById(itemId)
      .thenCompose(ItemsValidator::refuseWhenItemNotFound)
      .thenCompose(allowedStatusesValidator)
      .thenCompose(this::updateRequestStatusIfRequired)
      .thenApply(item -> item.changeStatus(targetStatus))
      .thenCompose(itemCollection::update);
  }
```

And then for the `Unknown` it will be:
```java
public CompletableFuture<Item> processMarkItemUnknown(WebContext context) {
   return moveItemIntoStatus(context, UNKNOWN, MarkAsUnknownValidators::itemHasAllowedStatusToMarkAsUnknown);
```